modules/jfx.incubator.input/src/main/java/jfx/incubator/scene/control/input/KeyBinding.java

```diff
@@ -0,0 +1,787 @@
+    }
+
+    /**
+     * Creates a KeyBinding which corresponds to the key press with the specified {@code KeyCode}
+     * and the {@code alt} key modifier ({@code option} on macOS).
+     * <p>
+     * This method is equivalent to {@link #option(KeyCode)} on macOS.
+     *
+     * @param code the key code
+     * @return the KeyBinding
+     */
+    public static KeyBinding alt(KeyCode code) {
+        return create(code, KCondition.KEY_PRESSED, KCondition.ALT);
+    }
+
+    /**
+     * Creates a KeyBinding which corresponds to the key press with the specified {@code KeyCode}
+     * and the {@code ctrl} key modifier ({@code control} on macOS).
+     *
+     * @param code the key code
+     * @return the KeyBinding
+     */
+    public static KeyBinding ctrl(KeyCode code) {
+        return create(code, KCondition.KEY_PRESSED, KCondition.CTRL);
+    }
+
+    /**
+     * Creates a KeyBinding which corresponds to the key press with the specified {@code KeyCode}
+     * and the {@code shift} + {@code ctrl} key modifier ({@code control} on macOS).
+     *
+     * @param code the key code
+     * @return the KeyBinding
+     */
+    public static KeyBinding ctrlShift(KeyCode code) {
+        return create(code, KCondition.KEY_PRESSED, KCondition.CTRL, KCondition.SHIFT);
+    }
+
+    /**
+     * Creates a KeyBinding which corresponds to the key press with the specified {@code KeyCode}
+     * and the {@code option} key modifier on macOS.
+     * <p>
+     * This method returns {@code null} on non-macOS platforms.
+     *
+     * @param code the key code
+     * @return the KeyBinding, or null
+     */
+    public static KeyBinding option(KeyCode code) {
+        return create(code, KCondition.KEY_PRESSED, KCondition.OPTION);
+    }
+
+    /**
+     * Creates a KeyBinding which corresponds to the key press with the specified {@code KeyCode}
+     * and the {@code shift} key modifier.
+     *
+     * @param code the key code
+     * @return the KeyBinding
+     */
+    public static KeyBinding shift(KeyCode code) {
+        return create(code, KCondition.KEY_PRESSED, KCondition.SHIFT);
+    }
+
+    /**
+     * Creates a KeyBinding which corresponds to the key press with the specified {@code KeyCode}
+     * and the shortcut key modifier ({@code ⌘ command} on macOS, {@code ctrl} elsewhere).
+     *
+     * @param code the key code
+     * @return the KeyBinding
+     */
+    public static KeyBinding shortcut(KeyCode code) {
+        return create(code, KCondition.KEY_PRESSED, KCondition.SHORTCUT);
+    }
+
+    /**
+     * Creates a KeyBinding which corresponds to the key press with the specified {@code KeyCode}
+     * and the {@code shift} + {@code option} key modifiers on macOS.
+     * <p>
+     * This method returns {@code null} on non-macOS platforms.
+     *
+     * @param code the key code
+     * @return the KeyBinding, or null
+     */
+    public static KeyBinding shiftOption(KeyCode code) {
```

Comment:
I haven't needed shift-alt, but I suppose it can be added for other platforms.  shift-option, on the other hand is used on mac (in the TextInputControl hierarchy, for example).